### ISSUE TITLE
Complex project.clj files are problematic

### ISSUE THREAD

**Issue by nasser** [arcadia-unity/Arcadia#244]
Lein coordinates that use keyword arguments fail our specs.

Example: [`tools.analyzer`](https://github.com/clojure/tools.analyzer/blob/master/project.clj#L9)

```
System.Exception: invalid thing:
==================================================
In:  [0]
val: com.datomic/datomic-free
fails
 spec::arcadia.packages.data/group
 at: [:arcadia.packages.data/vector-dependency
      :long
      :arcadia.packages.data/group]
predicate: 
     string?
furthermore:
     :path  [:arcadia.packages.data/vector-dependency :long :arcadia.packages.data/group]
```

**Comment by timsgardner** [comment]
easy enough to fix, just need to loosen the spec and make sure the rest of the logic is ok with symbols

**Comment by timsgardner** [comment]
(though I bet datomic-free won't compile for us anyway)

**Comment by timsgardner** [comment]
Need to make sure the new policy makes sense with new NuGet story

**Comment by sogaiu** [comment]
With specter (modified for arcadia use) cloned into some-project/Assets/, the following sort of thing shows up in the console messages:

```
Encountered System.Exception exception for loadpath extension function keyed to :arcadia.internal.leiningen/loadpath-fn during computation of loadpath.System.Exception: Unsupported file <project>/Assets/specter/project.clj: '#' found
```

Specter's project.clj has some '#' characters in it:

```
            {#"target/classes" "https://github.com/nathanmarz/specter/tree/{version}/src/clj/{classpath}x#L{line}"
             #".*"             "https://github.com/nathanmarz/specter/tree/{version}/src/clj/{classpath}#L{line}"}}
```

https://github.com/nathanmarz/specter/blob/fe1deb51951f892f7dac29b6d65992afd2c94651/project.clj#L20-L21

May be the code responsible is the following:

https://github.com/arcadia-unity/Arcadia/blob/24b43b946d3ff6909788034dd8c8dd88c7ff8112/Source/arcadia/internal/leiningen.clj#L54-L55

OS: Manjaro Linux
Unity: 2018.3.0f2
Arcadia: b713ffebf781743752f369878a432b190630f552 (beta)

**Comment by sogaiu** [comment]
Another item from specter's project.clj file -- it currently begins with:

```
(def VERSION (.trim (slurp "VERSION")))
```

https://github.com/nathanmarz/specter/blob/fe1deb51951f892f7dac29b6d65992afd2c94651/project.clj#L1

IIUC, one consequence is that currently Arcadia doesn't end up with anything useful from the project.clj file -- for specter, this means the corresponding source-paths doesn't get filled in so the load path doesn't point to an appropriate location.

For this specific case, possibly just parsing the defproject (and ignoring the def) would be sufficient?

**Comment by nasser** [comment]
Fixed in 771683ae3b97adb614bb5b28531ab166b88fd74c
